server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -152,6 +154,28 @@ public TaskManager(Settings settings, ThreadPool threadPool, Set<String> taskHea
         taskResourceConsumer = new HashSet<>();
     }
 
+    /**
+     * Listener that gets invoked when a task gets cancelled.
+     */
+    public interface TaskCancellationListener {
+        void onTaskCancelled(CancellableTask task);
+    }
+
+    public void addTaskCancellationListeners(TaskCancellationListener taskCancellationListener) {
+        this.taskCancellationListeners.add(taskCancellationListener);
+    }
+
+    /**
+     * Listener that gets invoked when a task completes.
+     */
+    public interface TaskCompletionListener {
+        void onTaskCompleted(Task task);
+    }
+
+    public void addTaskCompletionListener(TaskCompletionListener listener) {
+        this.taskCompletionListeners.add(listener);
+    }
+
```

Comment:
Make sense. Will do.

Reply:
@Bukhtawar Just thinking again, I think it is better to have these listeners defined separately?
As merging into one, we will need to create a single ```List<TaskEventListeners>``` for others to register. And with interface like below
```
 interface TaskEventListeners {
        void onTaskCancelled(CancellableTask task);
        void onTaskCompleted(Task task);
    }
```

As in case someone only need to listen to cancellation events, they will also be invoked for task completion in above case which seems wrong. Let me know what you think.

Reply:
You can make it Optional for implementers to override handling of specific events
```
interface TaskEventListeners {
        default void onTaskCancelled(CancellableTask task) {};
        default void onTaskCompleted(Task task) {};
    }

Reply:
Done. Merged these two listeners into one.